### PR TITLE
OCLOMRS-475: While editing a mapping when the source changes clear all the other fields in the row

### PR DESCRIPTION
**Pull request by alexkayabula**
# JIRA TICKET NAME:
[While editing a mapping when the source changes clear all the other fields in the row.](https://issues.openmrs.org/browse/OCLOMRS-475)

# Summary:
When editing a mapping, clear all editable fields when the mappings' source is changed.

### CODE REVIEW

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 1]
```diff
@@ -252,6 +252,10 @@ export class EditConcept extends Component {
       if (modifyMap.url === url) {
         modifyMap[name] = value;
       }
+      if (name === 'source') {
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: Shouldn't we have a constant for this?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: I have replaced the expression with a constant.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 2]
```diff
@@ -247,11 +247,16 @@ export class EditConcept extends Component {
   updateEventListener = (event, url) => {
     const { value, name } = event.target;
     const { mappings } = this.state;
+    const eventName = name === 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: In how many places are using "source"?

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx [hunk 3]
```diff
@@ -402,10 +402,27 @@ describe('Test suite for mappings on existing concepts', () => {
...
     const url = '1234';
+    const { value } = event.target;
     const instance = wrapper.find('EditConcept').instance();
     instance.updateEventListener({ target: { tabIndex: 0, name: INTERNAL_MAPPING_DEFAULT_SOURCE, value: '' } });
     instance.updateEventListener(event, url);
-    expect(instance.state.mappings[0].to_concept_code).not.toEqual(null);
+    expect(instance.state.mappings[0].to_concept_code).toEqual(value);
+  });
+
+  it('should call updateEventListener function on source change', () => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: How is this related to the ticket functionality?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 2 above)
Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: I am using it in one place.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: Are you sure?

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx (hunk 3 above)
Comment (alexkayabula) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: It is meant to test for the event where a different source of a mapping is changed, the value of the "to_concept_code" is cleared because "to_concept_code" is equal to " ".

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 2 above)
Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: Yes I am.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: That is not true

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx (hunk 3 above)
Comment (dkayiwa) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: Can you do your tests in terms of end user functionality?

Comment (alexkayabula) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: I have adjusted the test to stimulate user selection.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 2 above)
Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: I have double checked and I cannot find any other occurrence of that expression.

File: src/tests/dictionaryConcepts/container/EditConcept.test.jsx (hunk 3 above)
Comment (dkayiwa) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: Why should updateEventListener be called on source change?

Comment (alexkayabula) [reply src/tests/dictionaryConcepts/container/EditConcept.test.jsx]: This is due to the fact that whenever a change is made e.g when the source of the mapping changes, updateEventListener is called to update the state.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 2 above)
Comment (dkayiwa) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: Can you use a constant for it?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: I have changed it to a constant

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 4]
```diff
@@ -247,11 +247,16 @@ export class EditConcept extends Component {
   updateEventListener = (event, url) => {
     const { value, name } = event.target;
     const { mappings } = this.state;
+    const source = 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: Shouldn't you use a constant for the above?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: I have specified the constant in the helperFunction.js component.

File: src/components/dictionaryConcepts/components/helperFunction.js [hunk 5]
```diff
@@ -62,3 +62,4 @@ export const MAP_TYPE = {
 export const TRADITIONAL_OCL_HOST = urlConfig.TRADITIONAL_OCL_HOST;
 
 export const CUSTOM_SOURCE = 'Custom';
+export const SOURCE = 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/helperFunction.js]: Can you use a more descriptive name? What does SOURCE mean?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: I have given the constant a more descriptive name.

File: src/components/dictionaryConcepts/components/helperFunction.js [hunk 6]
```diff
@@ -62,3 +62,4 @@ export const MAP_TYPE = {
 export const TRADITIONAL_OCL_HOST = urlConfig.TRADITIONAL_OCL_HOST;
 
 export const CUSTOM_SOURCE = 'Custom';
+export const SELECTED_EVENT_NAME = 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/helperFunction.js]: Which kind or type of event is this?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: I have changed it to a more appropriate name.

File: src/components/dictionaryConcepts/components/helperFunction.js [hunk 7]
```diff
@@ -62,3 +62,4 @@ export const MAP_TYPE = {
 export const TRADITIONAL_OCL_HOST = urlConfig.TRADITIONAL_OCL_HOST;
 
 export const CUSTOM_SOURCE = 'Custom';
+export const SELECT_EVENT_NAME = 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/helperFunction.js]: Is this a JavaScript event named source?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: No, it is not, I have changed the name of the constant.

File: src/components/dictionaryConcepts/components/helperFunction.js [hunk 8]
```diff
@@ -62,3 +62,4 @@ export const MAP_TYPE = {
 export const TRADITIONAL_OCL_HOST = urlConfig.TRADITIONAL_OCL_HOST;
 
 export const CUSTOM_SOURCE = 'Custom';
+export const SELECT_NAME = 'source';
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/components/helperFunction.js]: What does select name mean?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: It is a constant that specifies the name attribute of the select tag.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/components/helperFunction.js]: Can you point me to this select tag and its name attribute? Do you mean that its name attribute value should be source?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: Yes, in the event a different source of a mapping is selected, the value of the name attribute is 'source'.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/components/helperFunction.js]: Can you point me to this select tag and its name attribute?

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: The select tag I meant is in the Create Mapping component.

Comment (dkayiwa) [reply src/components/dictionaryConcepts/components/helperFunction.js]: In that case ATTRIBUTE_NAME_SOURCE makes more sense.

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: Let me change it right away.

Comment (alexkayabula) [reply src/components/dictionaryConcepts/components/helperFunction.js]: I have changed the constant name to ATTRIBUTE_NAME_SOURCE.
